### PR TITLE
Consider group memberships in the OwnershipCard

### PR DESCRIPTION
**Pull request by dhenneke**
We tried the org-plugin and mentioned that the owned entities are not complete. I ~~export~~copied the `isOwnerOf` function from the catalog plugin and reuse it to also count entities that are owned by all groups the user is a member of.

New behavior:
* **User**: Count all entities that the user owns + the entities that all his groups own.
* **Group**: Count all the entities that the group directly owns. No child-group-owned entities are counted.

| Before (user guest)        | After (user guest)           |
| ------------- |-------------|
|       |  |

The `isOwnerOf` hook only works correctly for `User` entities today since Group-to-Group relationships are not modelled via `memberOf` but with `parentOf` relationships. Since I am not completely sure whether a group transitively owns all components of its children, I leave that for another PR.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by Rugvip — COMMENTED** [review]
Thanks!

Generally 👍 , but gonna bring around a discussion whether we want `isOwnerOf` to be exported and perhaps more importantly exported from the `catalog` plugin. It probably makes more sense to export it from the `@backstage/catalog-model` instead, so we can use it in both the frontend and backend. We also might want to establish some naming pattern or namespace for all similar functions, as I'm sensing there will be more of these :grin:

**Comment by dhenneke** [comment]
Agree, I went the most easy way here. Having it in catalog-model seems right. There will also be `isEntityNameEqual` or similar.

What probably really need here is a `useOwnedEntities(owner: Entity or EntityName)` hook that also loads the entities from the backend.

**Comment by freben** [comment]
Yes, let's avoid making new cross-plugin imports if we can help it. I'm in favor of putting it in `catalog-model`, in a file next to the relation definitions themselves (maybe `relationsLogic.ts` or something like that).

As for naming... That's never easy :) Any ideas?

**Comment by freben** [comment]
By the way, a hook like the one you mention - that could be a great candidate for `packages/catalog-client` could it not?

**Comment by dhenneke** [comment]
Yes, that seems reasonable. Would also open the door for doing server-side filtering in the future. Technically, it might also be good to do the aggregation that is done in this component in the backend, but I'm not sure if this is a commonly used function to provide a (typescript) interface for that.

**Comment by Rugvip** [comment]
@freben Would be unfortunate to have the `react` dependency in `catalog-client`

**Comment by Rugvip** [comment]
One possibility is to split up the `catalog` plugin into one piece that is the catalog browsing page, and another that contains only the entity pages and all utilities related to those. The second one would likely have a much more stable API and be depended on by other packages, while the first one would be pretty optional to use at all

**Comment by freben** [comment]
Oh I actually completely forgot it's an isomorphic package. Bah.

So... A new `catalog-react` package then? ;)

**Comment by dhenneke** [comment]
> @freben Would be unfortunate to have the `react` dependency in `catalog-client`

I'm not sure if it really needs to be a react hook in the `catalog-client`. It could just return a `Promise<Entity[]>` and let the frontend-consumers wrap it into a hook.

**Comment by dhenneke** [comment]
Can I unblock this PR somehow?

Proposal: I copy the `isOwnerOf` functionality to the orgs plugin to get this fixed. Afterwards I can create a new issue that collects existing requirements for common hooks or catalog-client-functions that acts as the place to discuss those. When everything is in place, we can come back and cleanup this plugin with the new interface(s).

**Comment by freben** [comment]
@dhenneke Oh for sure, don't get blocked on our unclear musings and thoughts here.

If you want, you can place `isOwnerOf` in `catalog-model`, possibly naming it something like `hasOwnerEntityRelation` to namespace and pin it down a bit, and then continue as proposed.

If you prefer to, and feel that it makes you move faster, localise it in the orgs plugin, and let us worry later about how to settle the ongoing ownership related discussions.

**Comment by Rugvip** [comment]
Yep completely fine to copy it over to the org plugin to not have to worry about that bit yet. Very much on the radar either way

**Comment by dhenneke** [comment]
I went with the copy-approach. I tried to implement the "all entities owned by a group (and all its child-groups)" logic but realised that the `parentOf` relations are not transitive but would require getting all the child-groups with the catalog-client. This is not possible in the `catalog-model` package, so it is better to handle this separately in a dedicated discussion.

**Comment by freben** [comment]
@dhenneke So it would work if we made `childOf` transitive (but not the other direction because it scales worse)... :) Hm. Since this is a common thing to want to query for, I'll give it some thought whether we can support it without complex nested queries.

**Comment by dhenneke** [comment]
@freben but would we need to make direct and transitive childs distinguishable? Looks a bit like the ascendants/descendants that was removed recently, just as relations (and one-sided) 😉

**Comment by dhenneke** [comment]
-> https://github.com/backstage/backstage/issues/3655

**Comment by freben** [comment]
Yeah that was indeed on my mind as we designed these... My thought was that we (or an organization themselves) can always add `directMemberOf` or, conversely, `directOrIndirectChildOf` etc.

**Review by freben — APPROVED** [review]

**Pull request merged**
